client/sse_test.go

```diff
@@ -95,10 +102,21 @@ func TestSSEMCPClient(t *testing.T) {
 
 		// Test ListTools
 		toolsRequest := mcp.ListToolsRequest{}
-		_, err = client.ListTools(ctx, toolsRequest)
+		toolListResult, err := client.ListTools(ctx, toolsRequest)
 		if err != nil {
 			t.Errorf("ListTools failed: %v", err)
 		}
+		if toolListResult == nil || len((*toolListResult).Tools) == 0 {
+			t.Errorf("Expected one tool")
+		}
+		testToolAnnotations := (*toolListResult).Tools[0].Annotations
```

Comment:
What prevents you from using proper asserts to check for the values?

Reply:
Align with the previous coding style.

Reply:
I think it is bad practice. On error you don't know what is wrong but understand you standpoint.